src/main/deltachat/chatlist.js

```diff
@@ -81,62 +83,64 @@ module.exports = class DCChatList extends SplitOut {
       chat.deaddrop = this._deadDropMessage(messageId)
     }
 
-    if (chat.id === C.DC_CHAT_ID_ARCHIVED_LINK) {
-      chat.isArchiveLink = true
-    }
-
+    console.log('getChatListItemsByIds', chatId)
     const summary = list.getSummary(i).toJson()
     const lastUpdated = summary.timestamp ? summary.timestamp * 1000 : null
 
     if (summary.text2 === '[The message was sent with non-verified encryption.. See "Info" for details.]') {
       summary.text2 = this._controller.translate('message_not_verified')
     }
-
+    const isGroup = isGroupChat(chat)
+    const contactIds = this._dc.getChatContacts(chatId)
     // This is NOT the Chat Oject, it's a smaller version for use as ChatListItem in the ChatList
     return {
       id: chat.id,
-      email: summary.text1,
-      name: chat.name,
+      // email: summary.text1,
+      name: chat.name || summary.text1,
       avatarPath: chat.profileImage,
-      color: chat.color,
+      color: integerToHexColor(chat.color),
       lastUpdated: lastUpdated,
       summary: {
         text1: summary.text1,
         text2: summary.text2,
         status: mapCoreMsgStatus2String(summary.state)
       },
       deaddrop: chat.deaddrop,
-      contacts: chat.contacts,
+      // contacts: chat.contacts,
       isVerified: chat.isVerified,
-      isGroup: chat.isGroup,
-      freshMessageCounter: chat.freshMessageCounter,
-      isArchiveLink: chat.isArchiveLink,
-      selfInGroup: chat.selfInGroup
+      isGroup: isGroup,
+      freshMessageCounter: this._dc.getFreshMessageCount(chatId),
+      isArchiveLink: chat.id === C.DC_CHAT_ID_ARCHIVED_LINK,
+      selfInGroup: isGroup && contactIds.indexOf(C.DC_CONTACT_ID_SELF) !== -1
     }
   }
 
-  getFullChatById (chatId, loadMessages) {
+  __getChatById (chatId) {
     if (!chatId) return null
     const rawChat = this._dc.getChat(chatId)
     if (!rawChat) return null
-    this._controller._pages = 0
-    const chat = rawChat.toJson()
+    return rawChat.toJson()
+  }
+
+  __getDraft (chatId) {
     const draft = this._dc.getDraft(chatId)
+    return draft ? draft.getText() : ''
+  }
 
-    if (draft) {
-      chat.draft = draft.getText()
-    } else {
-      chat.draft = ''
-    }
+  getFullChatById (chatId, loadMessages) {
+    const chat = this.__getChatById(chatId)
+    if (chat === null) return null
+    this._controller._pages = 0
```

Comment:
why is a global var change in a getter function? I know its a parameter for `this._controller.messageList._messagesToRender(messageIds)` but saving it globally seems not so good practice in my eyes.

Reply:
Which variable is global? I know that we somewhere have a hack to save the chatlist object somewhere as a global variable because we otherwise need to refetch it again and again.

Reply:
`this._controller._pages` global relativ to the dc contoller

Reply:
yup this is ugly code. This whole stuff is near a refactoring, i want to have a method getChatListItem and getMessages. Currently it's spread over various places and reuses code/fucntions because of the old state architecture.